packages/progress-bar/src/vaadin-progress-bar-base.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright (c) 2017 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { css } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+
+export const progressBarStyles = css`
+  :host {
+    display: block;
+    width: 100%; /* prevent collapsing inside non-stretching column flex */
+    height: 8px;
+  }
+
+  :host([hidden]) {
+    display: none !important;
+  }
+
+  [part='bar'] {
+    height: 100%;
+  }
+
+  [part='value'] {
+    height: 100%;
+    transform-origin: 0 50%;
+    transform: scaleX(var(--vaadin-progress-value));
+  }
+
+  :host([dir='rtl']) [part='value'] {
+    transform-origin: 100% 50%;
+  }
+`;
+
+export const progressBarTemplate = (html) => html`
+  <div part="bar">
+    <div part="value"></div>
+  </div>
+`;
```

Comment:
nit: The template is so simple, this is probably not worth the hacky mechanism & return type. Not a blocker.

Reply:
Good point, thanks. Updated and renamed the PR accordingly.